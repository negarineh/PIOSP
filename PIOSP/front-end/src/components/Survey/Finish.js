import React from 'react';
import PropTypes from 'prop-types';

import '../stylesheets/Survey.css';
import '../stylesheets/MainPage.css';
import '../stylesheets/Survey.css';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Segment, Step} from 'semantic-ui-react';
import {Link} from 'react-router-dom';


// last page of survey 
class Finish extends React.Component {
    constructor(props) {
      super(props);   
      this.state={
        stepIndex: 0,
      };
    }
    
  render() {
 
  return (
    <div className="mainpage">
     <MuiThemeProvider>
      {/* <Header/> */}
      <div className="wrappermainpage">      

      <div style={{width: 'auto', margin: 'auto', color:"dimgray", paddingLeft:"10px", overflow:'auto'}}>
      <Step.Group ordered size="mini" >
              <Step completed >
                <Step.Content>
                  <Step.Title>Enrolling</Step.Title>
                  <Step.Description>Insert your email </Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Confirmed Email </Step.Title>
                  <Step.Description>Welcome to our Survey</Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Survey Questions</Step.Title>
                  <Step.Description>Please answer questions</Step.Description>
                </Step.Content>
              </Step>

              <Step completed >
                <Step.Content>
                  <Step.Title>Rating and Further Experiences</Step.Title>
                  <Step.Description>Please rate our survey</Step.Description>
                </Step.Content>
              </Step>

              <Step active >
                <Step.Content>
                  <Step.Title>Survey Submitted</Step.Title>
                  <Step.Description>Thanks for your participation</Step.Description>
                </Step.Content>
              </Step>
            </Step.Group>
        </div>
        <Segment style={{margin:"10px", overflow:"auto"}}>
            <div style={{textAlign:'center'}}>
              <p key="a" className="psurvey21" > 
              Thank for your participation
              </p>
              
              <button 
                          type="submit" 
                          className="btn btn-primary2100"
                          
                          >
                  {/* Survey Home Page */}
                <Link  to={"/mainpage"}>Survey Home Page</Link>
              </button>
            </div>
        </Segment>
      </div>
      </MuiThemeProvider>
    </div>
      );
    }
  }

  Finish.defaultProps = {
    history: {},
  }

// prop checks
  Finish.propTypes = {
    history: PropTypes.object.isRequired,
};

export default Finish;