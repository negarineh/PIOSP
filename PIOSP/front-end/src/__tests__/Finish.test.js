import React from "react";
import Enzyme, { shallow } from "enzyme";
import assert from "assert";
import Finish from "../components/Survey/Finish";
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
 
Enzyme.configure({ adapter: new Adapter() });

// unit tests for the UserDeletePrompt component
describe('UserDeletePrompt component', () => {
  describe('render()', () => {
    it('should render the component', () => {
      const props = {history: {}};
      const wrapper = shallow(<Finish {...props}/>);
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      assert.equal(wrapper.length, 1);
    });
  });
});
