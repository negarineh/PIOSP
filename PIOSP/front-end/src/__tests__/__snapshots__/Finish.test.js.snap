// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDeletePrompt component render() should render the component 1`] = `
<div
  className="mainpage"
>
  <MuiThemeProvider>
    <div
      className="wrappermainpage"
    >
      <div
        style={
          Object {
            "color": "dimgray",
            "margin": "auto",
            "overflow": "auto",
            "paddingLeft": "10px",
            "width": "auto",
          }
        }
      >
        <StepGroup
          ordered={true}
          size="mini"
        >
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Enrolling
              </StepTitle>
              <StepDescription>
                Insert your email 
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Confirmed Email 
              </StepTitle>
              <StepDescription>
                Welcome to our Survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Survey Questions
              </StepTitle>
              <StepDescription>
                Please answer questions
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            completed={true}
          >
            <StepContent>
              <StepTitle>
                Rating and Further Experiences
              </StepTitle>
              <StepDescription>
                Please rate our survey
              </StepDescription>
            </StepContent>
          </Step>
          <Step
            active={true}
          >
            <StepContent>
              <StepTitle>
                Survey Submitted
              </StepTitle>
              <StepDescription>
                Thanks for your participation
              </StepDescription>
            </StepContent>
          </Step>
        </StepGroup>
      </div>
      <Segment
        style={
          Object {
            "margin": "10px",
            "overflow": "auto",
          }
        }
      >
        <div
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          <p
            className="psurvey21"
            key="a"
          >
            Thank for your participation
          </p>
          <button
            className="btn btn-primary2100"
            type="submit"
          >
            <Link
              replace={false}
              to="/mainpage"
            >
              Survey Home Page
            </Link>
          </button>
        </div>
      </Segment>
    </div>
  </MuiThemeProvider>
</div>
`;
